docassemble/ALToolbox/al_income.py

```diff
@@ -0,0 +1,1015 @@
+            ("cd", "Certificate of Deposit"),
+            ("ira", "Individual Retirement Account"),
+            ("mutual fund", "Money or Mutual Fund"),
+            ("stocks", "Stocks or Bonds"),
+            ("trust", "Trust Fund"),
+            ("checking", "Checking Account"),
+            ("vehicle", "Vehicle"),
+            ("real estate", "Real Estate"),
+            ("other", "Other Asset"),
+        ]
+    )
+    return source_list
+
+
+def al_income_source_list():
+    """Returns a dict of income sources for a multiple choice dropdown."""
+    source_list = DAOrderedDict()
+    source_list["wages"] = "A job or self-employment"
+
+    source_list.elements.update(al_non_wage_income_list())
+    source_list.auto_gather = False
+    source_list.gathered = True
+
+    return source_list
+
+
+def al_non_wage_income_list():
+    """Returns a dict of income sources, excluding wages."""
+    source_list = DAOrderedDict()
+    source_list.auto_gather = False
+    source_list.gathered = True
+    source_list.elements.update(
+        [
+            ("SSR", "Social Security Retirement Benefits"),
+            ("SSDI", "Social Security Disability Benefits"),
+            ("SSI", "Supplemental Security Income (SSI)"),
+            ("pension", "Pension"),
+            ("TAFDC", "TAFDC"),
+            ("public assistance", "Other public assistance"),
+            ("SNAP", "Food Stamps (SNAP)"),
+            ("rent", "Income from real estate (rent, etc)"),
+            ("room and board", "Room and/or Board Payments"),
+            ("child support", "Child Support"),
+            ("alimony", "Alimony"),
+            ("other support", "Other Support"),
+            ("other", "Other"),
+        ]
+    )
+    return source_list
+
+
+def al_expense_source_list():
+    """Returns a dict of expense sources for a multiple choice dropdown."""
+    source_list = DAOrderedDict()
+    source_list.auto_gather = False
+    source_list.gathered = True
+    source_list.elements.update(
+        [
+            ("rent", "Rent"),
+            ("mortgage", "Mortgage"),
+            ("food", "Food"),
+            ("utilities", "Utilities"),
+            ("fuel", "Other Heating/Cooking Fuel"),
+            ("clothing", "Clothing"),
+            ("credit cards", "Credit Card Payments"),
+            ("property tax", "Property Tax (State and Local)"),
+            ("other taxes", "Other taxes and fees related to your home"),
+            ("insurance", "Insurance"),
+            ("medical", "Medical-Dental (after amount paid by insurance)"),
+            ("auto", "Car operation and maintenance"),
+            ("transportation", "Other transportation"),
+            ("charity", "Church or charitable donations"),
+            ("loan payments", "Loan, credit, or lay-away payments"),
+            ("support", "Support to someone not in household"),
+            ("other", "Other"),
+        ]
+    )
+    return source_list
+
+
+class ALIncome(PeriodicValue):
```

Comment:
What other methods are there? In the documentation, I only see are `.amount()`. The only other quirk I see is that the `currency()` function will return the value of the item, but I think that's an artifact of `.__str__()`. If not, I'm not sure how to affect it.

Maybe we want to avoid triggering the gathering of `.value` if `.exists` is false? I think that's a behavior from which we diverged.

Reply:
I don't think we use .exists as a pattern in any interviews that I built with the income module. The functionality of that is replaced with the .source attribute.

I'm okay with telling people to use `currency(some_value.total())` instead of just `currency(some_value)`. I didn't realize the `currency` function had that shortcut but I think it's not super discoverable.

We might be able to get similar functionality by defining a custom  `__float__` method.

Reply:
Based on this and the discussion in Slack, I'm adding just a `__str__` method, since `__float__` was described as an anti pattern.